### DISCUSSION TITLE
Confirmation that this is still codeable as of 09/30/2023

### DISCUSSION THREAD

**Discussion by aScHEABL** [adrianhajdin/project_web3.0#116] [General]
The previous post has been confirmed helpful, sadly the owner of this repo @adrianhajdin had chosen to delete the entire issues section altogether without prior notice.<br>
In order to help with this community and other fellow developers,<br>
I decided to reach out Github Support and I was able to get the original content down below.
![image](https://github.com/adrianhajdin/project_web3.0/assets/22256864/dd662bd7-e676-408d-90e1-8ead27c1ba37)
***
1. The Ropsten is deprecated, you could use Sepolia testnet instead, just replace everything Ropsten with Sepolia and it works just fine.

2. At round 1:00:00 (1 hour) into the video he said to install the devdependcies with commands, do not do this as you'll run into error later on when you run "npx hardhat test". If you have ran the command 
`npm install --save-dev hardhat @nomiclabs/hardhat-waffle ethereum-waffle chai @nomiclabs/hardhat-ethers ethers`
just delete the entire smart-contract folder and re-create it.
Then do this instead, just run "npx hardhat" (remember to cd into the folder) and it'll download all the necessary & latest packages for you, then you continue with the video until the part where you actually run "npx hardhat test", it'll show an error.
Watch the error carefully, it should print what packages you're missing, simply copy and paste and install the missing packages, remember to do it either "npm install --save-dev <package names>" or "yarn add --dev <package names>".
After you done this, the "npx hardhat test" command should run just fine.

There are some file names are different from the video, for instance: smart-contract/contracts/lock.sol
This shall not affect anything, you can continue with the video.

3. at around 2:00:00(2 hours) into the video, when you hit the submit button "Send Now" (the form and input fields), the browser will likely throw an error "no ethereum object" in the sendTransaction function in TransactionContext.jsx file.
that error is due to some changes in recent update of ethers package, so you have to go back to previous packages of ethers
to do that :
1) npm uninstall ethers or yarn remove ethers
2) npm i -S ethers@5.7.2 or yarn add --dev ethers@5.7.2

The above 3 steps should resolve all the problems you'll run into this project.

If you found this post helpful please respond your feedback and upvote this.

**Comment by tejAspire** [comment]
this very helpful
"no Ethereum object" this was my error i tried it solve  but i could not
thanks a lot for the solution
can u please give the solution for the latest version of ethters

**Reply by aScHEABL** [reply]
You cannot use the "latest" version of ethers as I've stated in my post above.
It has to be ethers@5.7.2 otherwise it'll fail.

**Comment by tejAspire** [comment]
how to disconnect and again connect the metamask account 
connect wallet button is not available ..
please help me out

**Reply by aScHEABL** [reply]
Probably something's wrong with your code, you didn't provide me either code or context, how should I help with nothing to see.

**Comment by Abhinayckrs** [comment]
Hey aScHEABL I am sharing my github link here where my codes are available 
I am using ethers at the version 6.9.0
But I have made changes according to that and till now have been successful in most of the thing but currently stuck at the video at 2:08:00

I am sharing my github link so that you can see the code and tell me 

Well The problem is that I am able to complete the transaction and it is being added to the blockchain as well and metamask is able to show the transaction

But the problem is that it doesn't even matter how much ever amount I put MetaMask is showing me you are sending 0 Sepolia.
Kindly help me

Here is the link for my github repo so that you could see this
https://github.com/Abhinayckrs/Web3.git

I tried debugging by myself and found that in client/src/context/TransactionContext.jsx line no 104 is not getting executed the console.log is throwing an error like this
contract runner does not support sending transactions (operation="sendTransaction", code=UNSUPPORTED_OPERATION, version=6.9.0)

Kindly look into this

**Comment by Sageikot001** [comment]
Hello @aScHEABL I have actually sort out a way to message you but none except here, LOL.

I have followed you above steps at about 1:31:45 where we try to deploy using sepolia, console shows "Compiled 1 Solidity file successfully (evm target: istanbul)." But immediately proceeds to show a TypeError rather than the address of the deployed contract. The error it shows is,

"TypeError: transactions.deployed is not a function
    at main (C:\Users\USER\Desktop\WEB DEVELOPMENT\REACT Projects\web-3\smart_contract\scripts\deploy.js:8:24)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at runMain (C:\Users\USER\Desktop\WEB DEVELOPMENT\REACT Projects\web-3\smart_contract\scripts\deploy.js:15:7)"
    
    I have not yet push code to github but for context the image below might help.
    
<img width="910" alt="Screenshot 2024-04-23 165958" src="https://github.com/adrianhajdin/project_web3.0/assets/117757201/4caa0616-868e-42c8-8fe5-9ba1bfd48774">

**Reply by Sageikot001** [reply]
I could really use some feedback on this as I am practically stuck for the time being.
